svsim/src/main/scala/Simulation.scala

```diff
@@ -5,14 +5,33 @@ package svsim
 import scala.collection.mutable.Queue
 import scala.util.{Success, Try}
 import java.io.{BufferedReader, BufferedWriter, File, InputStreamReader, OutputStreamWriter}
+import java.nio.file.{Path, Paths}
 
 final class Simulation private[svsim] (
   executableName:           String,
   settings:                 Simulation.Settings,
   val workingDirectoryPath: String,
   moduleInfo:               ModuleInfo
 ) {
-  private val executionScriptPath = s"$workingDirectoryPath/execution-script.txt"
+  private val executionScriptPath = s"$workingDirectory/execution-script.txt"
+
+  protected final def getProjectRootOrCwd: Path =
+    sys.props
+      .get("chisel.project.root")
+      .orElse(sys.env.get("MILL_WORKSPACE_ROOT"))
+      .orElse(sys.env.get("PWD"))
+      .map(Paths.get("").toAbsolutePath.resolve(_))
+      .getOrElse(Paths.get(""))
+      .toAbsolutePath
+
+  def workingDirectory: Path = {
```

Comment:
This looks like a better API to use generally should we deprecate `val workingDirectoryPath` and suggest people use this?

Reply:
I can't think why not, so I moved `getProjectRootOrCwd` to the `Simulation` object so it's accessable elsewhere (and public).
I also added `CHISEL_PROJECT_ROOT` env variable as an alternative way of specifying/detecting the project root.
Then I used it in `chisel3.testing.scalatest`'s implicit implementation of `HasTestingDirectory`.
Now `scalatest.ChiselSim` build directory in my mill projects are created in the workspace folder (`./build`)  instead of being hidden under `./out/module_name/tests/testOnly.dest/sandbox/build`.